lib/elixir/lib/keyword.ex

```diff
@@ -186,6 +186,142 @@ defmodule Keyword do
     :lists.foldl(fun, [], Enum.reverse(pairs))
   end
 
+  @doc """
+  Ensures the first argument is a `keyword` with the given
+  keys and default values.
+
+  The second argument must be a list of atoms, specifying
+  a given key, or tuples specifying a key and a default value.
+
+
+  It returns `{:ok, keyword}` in case the keyword list only has
+  the given keys with returns the validated keyword with default
+  values applied. Otherwise it returns `{:error, invalid_keys}`
+  with invalid keys.
```

Comment:
```suggestion

  If the keyword list has only the given keys, it returns
  `{:ok, keyword}` with default values applied. Otherwise it
  returns `{:error, invalid_keys}` with invalid keys.
```